src/cpp/session/modules/SessionHTMLPreview.cpp

```diff
@@ -964,6 +975,46 @@ void handlePreviewRequest(const http::Request& request,
    }
 }
 
+SEXP rs_showPageViewer(SEXP fileSEXP)
+{
+   try
+   {
+      // get full path to file
+      FilePath filePath = module_context::resolveAliasedPath(r::sexp::safeAsString(fileSEXP));
+
+      // create temp file to write standalone html to (place it within a temp dir
+      // so the default download file name is pretty)
+      FilePath viewerTempDir = module_context::tempFile("page-viewer", "dir");
+      Error error = viewerTempDir.ensureDirectory();
+      if (error)
+         throw r::exec::RErrorException(r::endUserErrorMessage(error));
+      FilePath viewerFilePath = viewerTempDir.childPath(filePath.filename());
+
+      // create base64 encoded version
+      error = module_context::createSelfContainedHtml(filePath, viewerFilePath);
```

Comment:
I don't think this will end up using a client event to send the data (the data is just fetched by the browser as normal). This wouldn't be any different than R Markdown previews we do today which I'd guess can be much larger due to embedded images.